src/stream.h

```diff
@@ -170,6 +170,7 @@ struct OutputBuffer {
 class MemoryStream : public Stream {
  public:
   WABT_DISALLOW_COPY_AND_ASSIGN(MemoryStream);
+  MemoryStream(MemoryStream&&) = default;
```

Comment:
Sigh... it's a little ugly.

- Pre-this change, CWriter has a single `MemoryStream func_stream_` member, which is an internal temporary stream that the whole function body gets temporarily written into (in `Write(const Func& func)`). This is used so that it can first scan the function body to learn the stack vars used (while writing the function body to the internal temporary stream), then write the stack var declarations to the real output, and then finally copy func_stream_ to the real output.
- Post-this change, CWriter stores a vector of temporary MemoryStreams (`std::vector<std::pair<std::string, MemoryStream>> func_sections_`), each tagged with the label of a caught exception. We do this because we don't know whether the exception is going to be rethrown until later (if we see a subsequent rethrow instruction that rethrows **that** caught exception), and I didn't want to emit the code to copy the exception to the stack unless it's really going to be rethrown. To put a MemoryStream in a vector, it needs to have a move constructor.

We could make this simpler at the cost of copying every caught exception to the stack (whether needed or not).